services/src/main/java/org/keycloak/exportimport/util/ExportUtils.java

```diff
@@ -489,40 +501,45 @@ public static UserRepresentation exportUser(KeycloakSession session, RealmModel
         }
 
         // Role mappings
-        Set<RoleModel> roles = user.getRoleMappings();
-        List<String> realmRoleNames = new ArrayList<>();
-        Map<String, List<String>> clientRoleNames = new HashMap<>();
-        for (RoleModel role : roles) {
-            if (role.getContainer() instanceof RealmModel) {
-                realmRoleNames.add(role.getName());
-            } else {
-                ClientModel client = (ClientModel)role.getContainer();
-                String clientId = client.getClientId();
-                List<String> currentClientRoles = clientRoleNames.get(clientId);
-                if (currentClientRoles == null) {
-                    currentClientRoles = new ArrayList<>();
-                    clientRoleNames.put(clientId, currentClientRoles);
+        if (options.isGroupsAndRolesIncluded()) {
```

Comment:
I think this might be out of scope of this PR. I'm not saying we shouldn't do this change, though. My only concern here is that this changes current behaviour and therefore theoretically breaks backward compatibility. And that's why this IMHO should be discussed in a separate JIRA/PR/ML thread.

Or am I missing something? Is this mandatory for the primary intention of this PR?

Reply:
Hi, good to hear from you! :-)
I think that this is not changing any current behavior because now users are never exported using the import/export from the web. Users are only exported in the boot process and in those exports roles and groups are always exported (everything is exported in the boot process).
Besides think that if you ask to export users but not groups/roles, users will have the group/role info for groups/roles that are not going to be exported (thing that I think it's wrong). IMHO this was done this way because this situation was never going to happen (if the users were exported, groups and roles were also exported, because that only happens in the boot export, so no check is necessary).
Let me know what you think.

Reply:
@rmartinc is this mandatory for this change? Otherwise, I'd agree with @vmuzikar and maybe we can have a follow up Jira for this. In this way we focus only on the bug fix.

Reply:
Yes, I think it's necessary. In the export page the user can select what to export (clients, groups & roles).  If the user selects clients but not groups&roles the service accounts should be exported but without groups and roles links (if not the info exported there cannot be re-imported, because the roles&groups will be missing). If you see in the comment before Stian says: "Service account users role/group mapping are exported only if export groups/roles is enabled".
The next point can be removed, although I prefer to maintain it because avoiding passwords is safer IMHO.